src/main/scala/introprog/IO.scala

```diff
@@ -139,23 +139,76 @@ object IO {
   }
 
     /**
-    * Loads an image from file.
+    * Load image from file.
     *
     * @param fileName the path the image that will be loaded.
-    * @return BufferedImage.
     * */
-  def loadImage(fileName: String): java.awt.image.BufferedImage = 
-    javax.imageio.ImageIO.read(new java.io.File(fileName))
-  
+  def loadImage(fileName: String): Option[Image] = 
```

Comment:
The problem with returning Option is that the user never gets to see what went wrong. We could use Either[Throwable, Image] but ]I'd rather return just an Image and let it just fail with exception for simplicitly. Then the user does not have to know about neither Either nor Option. And the user can always wrap it in a Try with the same effect as here.